.github/workflows/ansible-release.yml

```
@@ -0,0 +1,144 @@
+
+    steps:
+      - name: Check out antsibull
+        uses: actions/checkout@v3
+        with:
+          repository: ansible-community/antsibull
+          ref: main
+          path: antsibull
+
+      - name: Pre-create build directory
+        run: mkdir -p antsibull/build
+
+      # This is where the antsibull build-release role expects it by default
+      - name: Check out ansible-build-data under antsibull build directory
+        uses: actions/checkout@v3
+        with:
+          path: antsibull/build/ansible-build-data
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        working-directory: antsibull
+        run: |
+          python3 -m pip install ansible-core
+          python3 -m pip install antsibull
+          ansible-galaxy collection install community.general
+
+      - name: Checking out to a new branch
+        working-directory: antsibull/build/ansible-build-data
+        run: |
+          git checkout -b "publish-${ANSIBLE_VERSION}"
+
+      - name: Setting the user details
+        run: |
+          git config --global user.name "Github Actions"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+      # Run the playbook according to the current release process
+
+      - name: Building a release with the defaults
+        working-directory: antsibull
+        run: >-
+          ansible-playbook -vv playbooks/build-single-release.yaml
+          -e antsibull_data_reset=false
+          -e "antsibull_ansible_version=${ANSIBLE_VERSION}"
+
+        env:
+          # Make result better readable
+          ANSIBLE_CALLBACK_RESULT_FORMAT: yaml
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: sdist-and-wheel
+          path: antsibull/build/ansible-*.*
+
+      - name: Commit ansible-build-data and push the changes to github
+        working-directory: 'antsibull/build/ansible-build-data/${{ inputs.ansible-major-version }}'
+        run: |
+         git add *.rst *.build *.deps *.yaml *.in validate-tags-ignores
+         git commit -m "${{ env.CI_COMMIT_MESSAGE }}"
+         git push origin "publish-${ANSIBLE_VERSION}"
+
+      - name: Create PR to the ansible-build-data
+        working-directory: antsibull/build/ansible-build-data
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: >-
+          gh pr create
+          --base main
+          --head "publish-${{ inputs.ansible-version }}"
+          --title "${{ env.CI_COMMIT_MESSAGE }}"
+          --body "${{ env.CI_COMMIT_MESSAGE }}"
+
+  # publish job downloads the arifacts and publish it to PyPI
+
+  publish:
+    needs: [build]
+    name: 'Upload Ansible (${{ inputs.ansible-version }}) to PyPI'
+    runs-on: ubuntu-latest
+    environment:
+      name: pypirelease
+      url: https://pypi.org/project/ansible
```

Since we know the version at this point, let's make the URL more accurate:
```suggestion
      url: https://pypi.org/project/ansible/${{ inputs.ansible-version }}
```